File: backend/src/main/scala/domains/contest/objects/Contest.scala
package domains.contest.objects

import domains.user.objects.UserIdentity
import shared.objects.access.ResourceAccessPolicy

import java.time.Instant

/** 比赛聚合根，承载基础信息、赛题列表、访问策略、作者和审计时间。 */
final case class Contest(
  id: ContestId,
  slug: ContestSlug,
  title: ContestTitle,
  description: ContestDescription,
  startAt: Instant,
  endAt: Instant,
  problems: List[ContestProblemSummary],
  accessPolicy: ResourceAccessPolicy,
  author: Option[UserIdentity],
  createdAt: Instant,
  updatedAt: Instant
)

File: frontend/src/objects/contest/Contest.ts
import type { ContestDescription } from '@/objects/contest/ContestDescription'
import type { ContestId } from '@/objects/contest/ContestId'
import type { ContestProblemSummary } from '@/objects/contest/ContestProblemSummary'
import type { ContestSlug } from '@/objects/contest/ContestSlug'
import type { ContestTitle } from '@/objects/contest/ContestTitle'
import type { UserIdentity } from '@/objects/user/UserIdentity'
import type { ResourceAccessPolicy } from '@/objects/shared/access/ResourceAccessPolicy'

/** 比赛完整对象；包含时间窗口、题目列表、访问策略、作者和审计时间。 */
export type Contest = {
  id: ContestId
  slug: ContestSlug
  title: ContestTitle
  description: ContestDescription
  startAt: string
  endAt: string
  problems: ContestProblemSummary[]
  accessPolicy: ResourceAccessPolicy
  author: UserIdentity | null
  createdAt: string
  updatedAt: string
}
